### PR TITLE
Adding support for sovereign cloud

### PR DESCRIPTION
**Pull request by tauhid621**
Instead of hardcoded management url we will use "az cloud show" command to get the management url

### CODE REVIEW

File: __tests__/run.test.ts [hunk 1]
```diff
@@ -5,12 +5,14 @@ import { mocked } from 'ts-jest/utils';
...
 import * as fileHelper from '../src/utils/fileHelper'
 import * as client from '../src/utils/httpClient'
+import { ManagementUrlHelper } from '../src/auth/managementUrlHelper'
 
 const coreMock = mocked(core, true);
 const tokenGeneratorMock = mocked(tokenGenerator, true);
 const clientMock = mocked(client, true);
 const fileHelperMock = mocked(fileHelper, true);
 const scanHelperMock = mocked(scanHelper, true);
+const baseUrlMock = mocked(ManagementUrlHelper, true);
```
Comment (ajinkya599) [inline __tests__/run.test.ts:15]: Rename to `ManagementUrlHelperMock`?

File: src/auth/azAuthentication.ts [hunk 2]
```diff
@@ -17,7 +16,7 @@ export async function getAccessToken(): Promise<string> {
 export async function getAADToken(): Promise<string> {
   azPath = await io.which("az", true);
   return await getTokenFromAzCLI(
-    "account get-access-token --resource=" + resource
+    "account get-access-token --resource=" + await ManagementUrlHelper.getBaseUrl()
```
Comment (ajinkya599) [inline src/auth/azAuthentication.ts:19]: For better readability, store in a variable and use.

Comment (ajinkya599) [reply src/auth/azAuthentication.ts:19]: Suggest the same for other places too.

File: src/auth/managementUrlHelper.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        if(!this._baseUrl) {            
+            let azCloudDetails = JSON.parse(await ManagementUrlHelper.executeAzCliCommand('cloud show'));
+            const cloudEndpoints = azCloudDetails['endpoints'];
+            this._baseUrl = ManagementUrlHelper.getResourceManagerUrl(cloudEndpoints);
+        }
+
+        return this._baseUrl;
+    }
+    
+    public static async executeAzCliCommand(command: string, args?: string[]): Promise<string> {
```
Comment (ajinkya599) [inline src/auth/managementUrlHelper.ts:17]: This method looks duplicate of `getTokenFromAzCLI()` in `azAuthentication`. Suggestion:
- Rename `ManagementUrlHelper` -> `AzCli`
- Rename `getBaseUrl` -> `getManagementUrl`
- Rename `executeAzCliCommand` -> `executeCommand`
- Remove `getTokenFromAzCLI` from `azAuthentication` and use `executeCommand()`
- Rename `auth` folder to something else. `azure`?

File: src/azurePolicy/scanHelper.ts [hunk 4]
```diff
@@ -37,7 +38,7 @@ export async function triggerOnDemandScan(): Promise<ScanCompletionPoll[]> {
 }
 
 async function triggerScan(scope: string, token: string): Promise<string> {
-  let triggerScanUrl = `https://management.azure.com${scope}/providers/Microsoft.PolicyInsights/policyStates/latest/triggerEvaluation?api-version=2019-10-01`;
+  let triggerScanUrl = `${await ManagementUrlHelper.getBaseUrl()}${scope}/providers/Microsoft.PolicyInsights/policyStates/latest/triggerEvaluation?api-version=2019-10-01`;
```
Comment (ajinkya599) [inline src/azurePolicy/scanHelper.ts:41]: Not sure whether all environments support the API versions being used for the different APIs. Good to test E2E!

**Review by ajinkya599 — APPROVED** [review]

File: __tests__/run.test.ts (hunk 1 above)
Comment (tauhid621) [reply __tests__/run.test.ts]: Made AzCliMock

File: src/auth/azAuthentication.ts (hunk 2 above)
Comment (tauhid621) [reply src/auth/azAuthentication.ts]: File removed. Other places, variable used before adding to string

**Pull request merged**